app/src/main/java/com/example/game/activities/LoginActivity.java

```diff
@@ -0,0 +1,76 @@
+package com.example.game.activities;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.example.game.R;
+import com.example.game.databinding.ActivityLoginBinding;
+import com.google.android.material.snackbar.BaseTransientBottomBar;
+import com.google.android.material.snackbar.Snackbar;
+import com.parse.LogInCallback;
+import com.parse.ParseException;
+import com.parse.ParseUser;
+
+public class LoginActivity extends AppCompatActivity {
+    private ActivityLoginBinding binding;
+    private EditText etEmail;
+    private EditText etPassword;
+    private Button btnLogin;
+
+    public static final String  TAG = "LoginActivity";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        binding = ActivityLoginBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        etEmail = binding.etMail;
+        etPassword = binding.etPassword;
+        btnLogin = binding.btnLogin;
+
+        btnLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                //check whether the fields are not empty else login
+                String email = etEmail.getText().toString();
+                String password = etPassword.getText().toString();
+
+                if(email.isEmpty() || password.isEmpty()){
+                    Snackbar.make(btnLogin, "Email or password cannot be empty", BaseTransientBottomBar.LENGTH_SHORT);
+                }else{
+                    //Send Async request
+                    ParseUser.logInInBackground(email, password, new LogInCallback() {
+                        @Override
+                        public void done(ParseUser user, ParseException e) {
+                            if(e!= null){
+                                Log.e(TAG, "Error while logging in" + e);
+                                Snackbar.make(btnLogin, "Incorrect username or password. Try again.", BaseTransientBottomBar.LENGTH_SHORT);
+                            }else{
+                                Toast.makeText(LoginActivity.this, "Login Successful", Toast.LENGTH_SHORT).show();
+                                gotoMainActivity();
+                            }
+                        }
+                    });
+                }
+            }
+        });
+
+    }
+
+    private void gotoMainActivity() {
+        //go to MainActivity
+        Intent i = new Intent(LoginActivity.this, MainActivity.class);
+        startActivity(i);
+        finish();
+    }
+
+}
```

Comment:
nit:new line end of file

```suggestion
}

```